ir.c

```diff
@@ -1399,6 +1399,29 @@ bool ir_list_contains(const ir_list *l, ir_ref val)
 	return 0;
 }
 
+bool ir_remove_pis(ir_ctx *ctx)
+{
+	ir_ref n, i;
+	ir_insn *insn;
+
+	for (i = IR_UNUSED + 1, insn = ctx->ir_base + i; i < ctx->insns_count;) {
+		n = insn->inputs_count;
+
+		if (insn->op == IR_PI) {
+			ir_use_list_remove_all(ctx, insn->op2, i);
+			insn->op2 = IR_UNUSED;
+			insn->op = IR_COPY;
+			insn->inputs_count = 1;
```

Comment:
From the first look, I don't see a big reason in converting ``PI`` to ``COPY``.

``COPY`` is not going to be removed, and have to be handled by all the following passes. Actually, we may handle ``PI`` in the same way as ``COPY`` and keep the range info. On the other hand this will may additional overhead for all the following passes.

It's also possible to remove ``PI`` completely (turning it into NOP and re-connecting) input/def/use chains. T

Reply:
This may be done at the second phase of SCCP, that removes constant calculations and redundant copies.

I don't know what approach is better yet.